sources/packages/backend/apps/api/src/route-controllers/report/_tests_/e2e/report.institutions.controller.exportReport.e2e-spec.ts

```diff
@@ -0,0 +1,283 @@
+        auditUser,
+      },
+      { initialValues: { studyStartDate: "2010-09-01" } },
+    );
+    const firstSavedOffering = await db.educationProgramOffering.save(
+      firstFakeOffering,
+    );
+    // 2nd offering: with an application belonging to the location of the institution for which the report is generated.
+    const secondFakeOffering = createFakeEducationProgramOffering(
+      {
+        institution: collegeF,
+        institutionLocation: collegeFLocation,
+        auditUser,
+      },
+      { initialValues: { studyStartDate: "2010-09-01" } },
+    );
+    const secondSavedOffering = await db.educationProgramOffering.save(
+      secondFakeOffering,
+    );
+    await saveFakeApplication(
+      db.dataSource,
+      {
+        institution: collegeF,
+        institutionLocation: collegeFLocation,
+        student,
+        offering: secondSavedOffering,
+        programYear,
+      },
+      {
+        applicationStatus: ApplicationStatus.Completed,
+      },
+    );
+    // 3rd offering: with an application belonging to the location of the institution for which the report is not generated.
+    const thirdFakeOffering = createFakeEducationProgramOffering(
+      {
+        institution: collegeC,
+        institutionLocation: collegeCLocation,
+        auditUser,
+      },
+      { initialValues: { studyStartDate: "2010-09-01" } },
+    );
+    const thirdSavedOffering = await db.educationProgramOffering.save(
+      thirdFakeOffering,
+    );
+    await saveFakeApplication(
+      db.dataSource,
+      {
+        institution: collegeC,
+        institutionLocation: collegeCLocation,
+        student,
+        offering: thirdSavedOffering,
+        programYear,
+      },
+      {
+        applicationStatus: ApplicationStatus.Completed,
+      },
+    );
+    const payload = {
+      reportName: "Offering_Details_Report",
+      params: {
+        offeringIntensity: {
+          "Full Time": true,
+          "Part Time": false,
+        },
+        programYear: 10,
+      },
+    };
+    const dryRunSubmissionMock = jest.fn().mockResolvedValue({
+      valid: true,
+      formName: FormNames.ExportFinancialReports,
+      data: { data: payload },
+    });
+    formService.dryRunSubmission = dryRunSubmissionMock;
+    const endpoint = "/institutions/report";
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+    // Act/Assert
+
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .send(payload)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.CREATED)
+      .then((response) => {
+        const fileContent = response.request.res["text"];
+        const parsedResult = parse(fileContent, {
+          delimiter: ",",
+          header: true,
+          skipEmptyLines: "greedy",
+        });
+        expect(parsedResult.data).toEqual(
+          expect.arrayContaining([
+            expect.objectContaining({
```

Comment:
Are we using `expect.arrayContaining` and `expect.objectContaining` to allow the test to be executed multiple times and still pass without resetting the data?

Reply:
yes